cypress/support/commands.ts

```diff
@@ -1,37 +1,29 @@
 // https://on.cypress.io/custom-commands
 import 'cypress-wait-until';
 import 'cypress-file-upload';
-import 'cypress-xpath';
-import { v4 as uuid } from 'uuid';
+
+import { expect } from 'chai';
 
 // Logins the user given a username and password
 Cypress.Commands.add('login', ({ username, password }) => {
-  cy.visit('/');
-  cy.get('[data-login-btn]').click();
-  cy.get('[data-login-username]').type(username);
-  cy.get('[data-login-password]').type(password);
-  cy.get('[data-login-submit]').click();
-  cy.waitUntil(() =>
-    cy
-      .xpath('//*[@id="__BVID__11___BV_modal_header_"]/button')
-      .should('be.visible')
-      .click(),
-  );
+  cy.request(
+    `/api/user/login?usernameOrEmail=${username}&password=${password}`,
+  ).then((response) => {
+    expect(response.status).to.eq(200);
+    cy.reload();
+    cy.get('.username').should('have.text', username);
+  });
 });
 
-// Register a new user given a username and password. If they are not specified, a random one is generated and returned
-Cypress.Commands.add('register', ({ username, password }) => {
-  cy.visit('/');
-  cy.get('[data-login-btn]').click();
-  cy.get('[data-signup-username]').type(username);
-  cy.get('[data-signup-password]').type(password);
-  cy.get('[data-signup-repeat-password]').type(password);
-  cy.get('[data-signup-email]').type(`${username}@omegaup.com`);
-  cy.get('[data-signup-submit]').click();
-  cy.waitUntil(() =>
-    cy
-      .xpath('//*[@id="__BVID__11___BV_modal_header_"]/button')
-      .should('be.visible')
-      .click(),
-  );
-});
+// Register a new user given a username and password. If shouldLogin is true, then the user will be logged in after the registration
+Cypress.Commands.add(
+  'register',
+  ({ username, password, shouldLogin = false }) => {
```

Comment:
```suggestion
  ({ username, password }) => {
```

mejor dejemos que el que manda llamar esto haga el login. igual nada más es una línea.

aunque pensándolo bien, creo que lo ideal sería que _siempre_ hiciera login. no veo bajo qué situación no querríamos que eso ocurriera.